openhands/llm/retry_mixin.py

```diff
@@ -44,8 +43,6 @@ def retry_decorator(self, **kwargs):
 
     def log_retry_attempt(self, retry_state):
         """Log retry attempts."""
-        if should_exit():
-            raise OperationCancelled('Operation cancelled.')  # exits the @retry loop
```

Comment:
The exception is not used anymore, I think (in the app). We might want to keep it because it might be useful later, I don't know, we can also re-add it easily.

A little more curious is that a unit test is testing it, but it seems it succeeded in CI. 🤔

Reply:
The unit tests throws the error, then tests it is thrown - in practice it would throw whatever error caused the retry